File: dev-docs/snippets/blob-buckets/list-buckets.ts
import type { JsBaoClient } from "js-bao-wss-client";

// List every blob bucket for the current app (admin/owner only).
export async function listBuckets(client: JsBaoClient) {
  // #region example
  const buckets = await client.blobBuckets.listBuckets();
  // #endregion example
  return buckets;
}

File: dev-docs/snippets/blob-buckets/list-buckets.swift
import JsBaoClient

// List every blob bucket for the current app (admin/owner only).
// Swift returns `[BlobBucketInfo]` — each field is a typed property.
func listBuckets(client: JsBaoClient) async throws {
  // #region example
  let buckets = try await client.blobBuckets.listBuckets()
  // #endregion example
  _ = buckets
}
